base/system/services/services.c

```diff
@@ -107,6 +109,26 @@ CheckForLiveCD(VOID)
         ScmLiveSetup = TRUE;
     }
 
+    /* Read the SystemSetupInProgress value */
+    dwSize = sizeof(DWORD);
+    dwError = RegQueryValueExW(hSetupKey,
+                               L"SystemSetupInProgress",
+                               NULL,
+                               &dwType,
+                               (LPBYTE)&dwSetupInProgress,
+                               &dwSize);
+    if (dwError != ERROR_SUCCESS ||
+        dwType != REG_DWORD ||
+        dwSize != sizeof(DWORD) ||
+        dwSetupType == 0)
+        goto done;
+
+    if (dwSetupInProgress == 1)
```

Comment:
what if it's non-zero but not equal to one either?

Reply:
`SystemSetupInProgress` can never be above 1, it's either 1 (== TRUE the setup is in progress) or 0 (== FALSE). The key value is set 0 when the setup is complete. See all the use cases of this value -> https://git.reactos.org/?p=reactos.git&a=search&h=HEAD&st=grep&s=SystemSetupInProgress

I see no point on adding unnecessary validity checks as it's just unwanted bikeshedding really.